gossip/service/gossip_service.go

```diff
@@ -111,6 +111,7 @@ type gossipServiceImpl struct {
 	lock            sync.RWMutex
 	mcs             api.MessageCryptoService
 	peerIdentity    []byte
+	mspID           string
```

Comment:
you can extract the MSPID by taking the `peerIdentity` and passing into the `secAdv` that is right below this `mspID` field.

Reply:
Thanks, I was trying to find a way of extracting the mspID without having to pass it down the entire stack. Any other thoughts about this approach? I tried to ask @ale-linux but haven't heard back yet.

Reply:
Also that covers extracting the mspID for the gossip path -> coordinator. Is there an easier way of extracting from the membership update path via the `membershipInfoProvider`? I'd like to entirely do away with the `selfSignedData` in both paths since I think we're only using them for the existing policy evaluation logic, but I guess for now we can keep them for the sake of falling back in case the new method doesn't work.

Reply:
You can apply the same function of the `secAdv` to the Identity in the self signed data, of course

Reply:
That works for the gossip path since it has a `secAdv` but the `membershipInfoProvider` only takes an `IdentityDeserializerFactory` and `SignedData`